src/mon/Elector.cc

```diff
@@ -754,6 +755,16 @@ void Elector::notify_rank_removed(int rank_removed)
       }
     }
   }
+  // Case where we are removing the highest rank monitor, we only care 
+  // about when the rank is in dead_pinging, which means
+  // it was shutted down, so we can erase it from dead_pinging.
+  // We don't care in the case where rank_removed 
+  // is still in live_pinging since it can be added back since it is not dead.
```

Comment:
I think this PR deals with the case where you remove the monitor before shutting it down: https://github.com/ceph/ceph/pull/44993. It will remove the monitor from live_pinging when it detects that the monitor's rank is > ranks.size().

Reply:
I'll revise the the comment anyways so it is not confusing